catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/ImageAttribute.java

```diff
@@ -44,285 +37,440 @@
-            BasicTypes.INTEGER_TYPE),
-    IMAGE_ATTACHMENT_LEVEL("imageAttachmentLevel",
-            "IALVL",
-            ImageSegment::getAttachmentLevel,
-            BasicTypes.INTEGER_TYPE),
-    IMAGE_LOCATION("imageLocation",
+            BasicTypes.INTEGER_TYPE,
+            ATTRIBUTE_NAME_PREFIX);
+
+    public static final NitfAttribute<ImageSegment> IMAGE_ATTACHMENT_LEVEL =
+            new NitfAttributeImpl<>("imageAttachmentLevel",
+                    "IALVL",
+                    ImageSegment::getAttachmentLevel,
+                    BasicTypes.INTEGER_TYPE,
+                    ATTRIBUTE_NAME_PREFIX);
+
+    public static final NitfAttribute<ImageSegment> IMAGE_LOCATION = new NitfAttributeImpl<>(
+            "imageLocation",
             "ILOC",
             segment -> segment.getImageLocationRow() + "," + segment.getImageLocationColumn(),
-            BasicTypes.STRING_TYPE),
-    IMAGE_MAGNIFICATION("imageMagnification",
+            BasicTypes.STRING_TYPE,
+            ATTRIBUTE_NAME_PREFIX);
+
+    public static final NitfAttribute<ImageSegment> IMAGE_MAGNIFICATION = new NitfAttributeImpl<>(
+            "imageMagnification",
             "IMAG",
             segment -> segment.getImageMagnification()
                     .trim(),
-            BasicTypes.STRING_TYPE);
+            BasicTypes.STRING_TYPE,
+            ATTRIBUTE_NAME_PREFIX);
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ImageAttribute.class);
 
-    private static final String ATTRIBUTE_NAME_PREFIX = "image.";
-
-    private String shortName;
-
-    private String longName;
-
-    private Function<ImageSegment, Serializable> accessorFunction;
+    public static final NitfAttribute<ImageSegment> TARGET_IDENTIFIER =
+            new NitfAttributeImpl<>(Isr.TARGET_ID,
+                    "TGTID",
+                    ImageAttribute::getTargetId,
+                    new IsrAttributes().getAttributeDescriptor(Isr.TARGET_ID),
+                    "targetIdentifier",
+                    ATTRIBUTE_NAME_PREFIX);
 
-    private Set<AttributeDescriptor> attributeDescriptors;
+    private List<NitfAttribute<ImageSegment>> attributes;
 
-    ImageAttribute(final String lName, final String sName,
-            final Function<ImageSegment, Serializable> accessor, AttributeType attributeType) {
-        this.accessorFunction = accessor;
-        this.shortName = sName;
-        this.longName = lName;
-        // retrieving metacard attribute descriptor for this attribute to prevent later lookups
-        this.attributeDescriptors = Collections.singleton(new AttributeDescriptorImpl(
-                ExtNitfUtility.EXT_NITF_PREFIX + ATTRIBUTE_NAME_PREFIX + lName,
-                true, /* indexed */
-                true, /* stored */
-                false, /* tokenized */
-                true, /* multivalued */
-                attributeType));
-    }
+    public ImageAttribute() {
+        attributes = Arrays.asList(IMAGE_DATE_AND_TIME,
```

Comment:
Keeping these lists in-sync with the lists of constants defined in the classes is really going to be a maintenance nightmare and will eventually come back to bite us. We need to find a better way...